src/test/kotlin/io/github/nomisrev/routes/TagRouteSpec.kt

```diff
@@ -0,0 +1,63 @@
+package io.github.nomisrev.routes
+
+import arrow.core.flatMap
+import io.github.nefilim.kjwt.JWSHMAC512Algorithm
+import io.github.nefilim.kjwt.JWT
+import io.github.nomisrev.repo.UserId
+import io.github.nomisrev.service.CreateArticle
+import io.github.nomisrev.service.RegisterUser
+import io.github.nomisrev.withServer
+import io.kotest.assertions.arrow.core.shouldBeRight
+import io.kotest.assertions.arrow.core.shouldBeSome
+import io.kotest.core.spec.style.StringSpec
+import io.kotest.matchers.collections.shouldHaveSize
+import io.kotest.matchers.shouldBe
+import io.ktor.client.call.body
+import io.ktor.client.request.get
+import io.ktor.http.ContentType
+import io.ktor.http.HttpStatusCode
+import io.ktor.http.contentType
+
+class TagRouteSpec :
+  StringSpec({
+    // User
+    val validUsername = "username2"
+    val validEmail = "valid2@domain.com"
+    val validPw = "123456789"
+    // Article
+    val validTags = setOf("arrow", "ktor", "kotlin", "sqldelight")
+    val validTitle = "Fake Article Arrow "
+    val validDescription = "This is a fake article description."
+    val validBody = "Lorem ipsum dolor sit amet, consectetur adipiscing elit."
+
+    "Check for empty list retrieval" {
+      withServer {
+        val response = get("/tags") { contentType(ContentType.Application.Json) }
+
+        response.status shouldBe HttpStatusCode.OK
+        response.body<TagsResponse>().tags shouldHaveSize 0
+      }
+    }
+
+    "Can get all tags" {
+      withServer { dependencies ->
+        val userId =
+          dependencies.userService
+            .register(RegisterUser(validUsername, validEmail, validPw))
+            .flatMap { JWT.decodeT(it.value, JWSHMAC512Algorithm) }
+            .map { it.claimValueAsLong("id").shouldBeSome() }
+            .shouldBeRight()
+
+        dependencies.articleService
+          .createArticle(
+            CreateArticle(UserId(userId), validTitle, validDescription, validBody, validTags)
+          )
+          .shouldBeRight()
+
+        val response = get("/tags") { contentType(ContentType.Application.Json) }
+
+        response.status shouldBe HttpStatusCode.OK
+        response.body<TagsResponse>().tags shouldHaveSize 4
```

Comment:
**Nit:** can we check here `tags.toSet() shouldBe validTags`?

Reply:
I made a [commit](https://github.com/nomisRev/ktor-arrow-example/pull/176/commits/2c45f272689317e6f949dd18c572a3293aece19f) with this change. I can see the commit, but I can't see the change in the 'files changed' tab. Is this a repository configuration?

If we can consider this as finished, I will pick up another issue, if that's okay.

Reply:
Yes, that'd be great 🙌 . Not sure why "files changed" is not working correctly 🤔